analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -48,9 +50,14 @@ public RoslynSymbolicExecution(ControlFlowGraph cfg, SymbolicCheck[] checks)
         public void Execute()
         {
             // ToDo: Forbid running twice
+            var steps = 0;
             queue.Enqueue(new ExplodedNode(cfg.EntryBlock, ProgramState.Empty));
             while (queue.Any())
             {
+                if (steps++ > MaxStepCount)
+                {
+                    return;
```

Comment:
I think we should break the loop and not return. The execution was completed (earlier than planned but it's still completed) and we should send a message that this has happened.

Reply:
That could raise FPs that we need to avoid. As rules depending on `NotifyExecutionCompleted` (like the S2222, as a lucky example) needs to visit the sanitizing call.

Reply:
We also have rules which are fine with partial execution (e.g. NullDereference). Can't we return a result that lets the rules know how the execution is completed?

Reply:
Those rules will report their partial results on the fly and should not be affected. And this will actually be an improvement with the new engine, as the old engine also returns and doesn't reach the `OnExplorationEnded` part.

https://github.com/SonarSource/sonar-dotnet/blob/773f29259126499169fb5954bda5b7a44b1ce8d3/analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Sonar/AbstractExplodedGraph.cs#L87-L91